lnrpc/routerrpc/router_backend.go

```diff
@@ -103,7 +103,7 @@ type MissionControl interface {
 	// GetProbability is expected to return the success probability of a
 	// payment from fromNode to toNode.
 	GetProbability(fromNode, toNode route.Vertex,
-		amt lnwire.MilliSatoshi) float64
+		amt lnwire.MilliSatoshi, capacity btcutil.Amount) float64
```

Comment:
This is probably an edgy case, but what if the channel capacity is actually zero?

Reply:
Good question, is it even possible to have a channel with zero capacity, or do you refer to an error in capacity validation? In that case the capacity factor would not be active (it would be 1), so no change compared to the current situation. Either the channel has a `maxHTLCMsat` set, then it must be less than or equal to the capacity per spec. In that scenario it would be caught by the `amtInRange` check. The `amtInRange` check currently doesn't prohibit sending to a zero sat channel:
https://github.com/lightningnetwork/lnd/blob/e23c5dca7364bcd329559d86727d5e2e4a8bbcc3/routing/unified_policies.go#L104

In the worst-case scenario we would send to that route and would fail, avoiding the channel next time.